include/crocoddyl/core/actions/lqr.hxx

```diff
@@ -47,8 +48,27 @@ void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>
     data->xnext.noalias() += Fu_ * u;
     data->xnext += f0_;
   }
-  data->cost =
-      Scalar(0.5) * x.dot(Lxx_ * x) + Scalar(0.5) * u.dot(Luu_ * u) + x.dot(Lxu_ * u) + lx_.dot(x) + lu_.dot(u);
+
+  // cost = 0.5 * x^T*Lxx*x + 0.5 * u^T*Luu*u + x^T*Lxu*u + lx^T*x + lu^T*u
+  data->cost = Scalar(0.0);
+
+  // 0.5 * x^T*Lxx*x
+  d->Hx_tmp.noalias() = Lxx_ * x;
+  data->cost += Scalar(0.5) * x.transpose() * d->Hx_tmp;
+
+  // 0.5 * u^T*Luu*u
+  d->Hu_tmp.noalias() = Luu_ * u;
+  data->cost += Scalar(0.5) * u.transpose() * d->Hu_tmp;
+
+  // x^T*Lxu*u
+  d->Hx_tmp.noalias() = Lxu_ * u;
+  data->cost += x.transpose() * d->Hx_tmp;
+
+  // lx^T*x
+  data->cost += lx_.transpose() * x;
+
+  // lu^T*u
+  data->cost += lu_.transpose() * u;
```

Comment:
```suggestion
d->Lxx_x.noalias() = Lxx_ * x;
d->Luu_x.noalias() = Luu_ * u;
data->cost = Scalar(0.5) * x.dot(d->Lx_x);
data->cost += Scalar(0.5) * x.dot(d->Lx_u);
d->Lxx_x.noalias() = Lxu_ * u;
data->cost += Scalar(0.5) * x.dot(d->Lx_x);
data->cost += lx_.dot(x);
data->cost += lu_.dot(u);
```

Additionally, I think we could add some `dot` in the same line. I think this should not allocate memory as the results is a scalar. Am I right?

Reply:
Removed comments and amended with dot